javascript/ql/src/Security/CWE-400/PrototypePollutionUtility.ql

```diff
@@ -91,7 +182,14 @@ predicate dynamicPropWrite(DataFlow::Node base, DataFlow::Node prop, DataFlow::N
     // Prune writes that are unlikely to modify Object.prototype.
     // This is mainly for performance, but may block certain results due to
     // not tracking out of function returns and into callbacks.
-    isPotentiallyObjectPrototype(base.getALocalSource())
+    isPotentiallyObjectPrototype(base.getALocalSource()) and
+
+    // Ignore writes with an obviously safe RHS.
+    not exists(Expr e | e = rhs.asExpr() |
+      e instanceof Literal or
+      e instanceof ObjectExpr or
+      e instanceof ArrayExpr
+    )
```

Comment:
```suggestion
    isPotentiallyObjectPrototype(base.getALocalSource())
```

Do you need this RHS check? 
In `isPrototypePollutingAssignment` you are already sure that `rhs` is either a `ParameterNode` or has flow from an enumerated property name, so I don't see this check affecting the flagged results.  
(This change does change the result from the `nodes` and `edges` predicates). 

(Also, I wanted to test out multi-line suggestions :tada:)

Reply:
It's a pruning step to remove uninteresting sinks early on.